addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/wso2iotsHandler.java

```diff
@@ -0,0 +1,334 @@
+                    }
+                }
+            };
+
+            int delay = config.refresh.intValue();
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    /**
+     * Update the channel from the last building monitor data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    private void updateChannel(String channelId, BigDecimal iotsResponse) {
+        if (isLinked(channelId)) {
+            State state = null;
+            if (iotsResponse == null) {
+                state = UnDefType.UNDEF;
+            } else if (iotsResponse instanceof BigDecimal) {
+                state = new DecimalType(iotsResponse);
+
+            } else {
+                logger.warn("Update channel {}: Unsupported value type {}", channelId,
+                        iotsResponse.getClass().getSimpleName());
+            }
+            logger.debug("Update channel {} with state {} ({})", channelId, (state == null) ? "null" : state.toString(),
+                    (iotsResponse == null) ? "null" : iotsResponse.getClass().getSimpleName());
+
+            // Update the channel
+            if (state != null) {
+                updateState(channelId, state);
+            }
+        }
+    }
+
+    private BigDecimal updateBuildingMonitorData(String sensorType) throws Exception {
+
+        BigDecimal iotsResponse = null;
+        String errormsg = null;
+        String response1 = null;
+        wso2iotsConfiguration config = getConfigAs(wso2iotsConfiguration.class);
+        String Access_Token = config.apikey;
+        String id = config.deviceId;
+        long time = System.currentTimeMillis() / 1000;
+        String fromTime = String.valueOf(time - API_RESPONSE_INTERVAL);
+        String toTime = String.valueOf(time);
+
+        String url = "https://localhost:8243/senseme/device/1.0.0/stats/";
+        url = url + id + "?from=" + fromTime + "&to=" + toTime + "&sensorType=" + sensorType;
+
+        URL obj;
+
+        obj = this.validateCertificate(url);
+
+        if (obj == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Certificate validation failed");
+            return null;
+        } else {
+
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+
+            con.setRequestMethod("GET");
+            con.setRequestProperty("Accept", "application/json");
+            con.setRequestProperty("Authorization", "Bearer " + Access_Token);
+            try {
+
+                BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
+                String inputLine;
+                StringBuffer response = new StringBuffer();
+
+                while ((inputLine = in.readLine()) != null) {
+                    response.append(inputLine);
+                }
+                in.close();
```

Comment:
You should put the reader in the try -> `try (BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));) {` Then this close can be removed. It's also saver because the close should be called in the finally block.